tests/test_handlers.py

```diff
@@ -1231,3 +1224,27 @@ def loss_fn2(bcc_params):
 
     assert loss_fn(params) == loss_fn2(params)
     np.testing.assert_array_equal(grad_fn(params), grad_fn2(params))
+
+    # now check for correctness on the full protein set
+    # if the initial charges are the same, then the final charges should
+    # also be the same by symmetry
+    atom_idx_to_res_name = {}
+    cur_atom = 0
+    for i in range(len(pbcc.template_for_residue)):
+        tfr = pbcc.template_for_residue[i]
+        n_atoms = len(tfr.atoms)
+        for j in range(n_atoms):
+            atom_idx_to_res_name[cur_atom + j] = f"{tfr.name}"
+        cur_atom += n_atoms
+
+    # charges that are the same and belong to the same residue should
+    # be the same after applying the corrections
+    init_q_params = pbcc.initial_charges
+    sym_charge_idxs = defaultdict(list)
+    for i, q in enumerate(init_q_params):
+        sym_charge_idxs[(q, atom_idx_to_res_name[i])].append(i)
+
+    for k, group in sym_charge_idxs.items():
+        first = group[0]
+        for other in group[1:]:
+            np.testing.assert_almost_equal(final_q_params[other], final_q_params[first])
```

Comment:
Are there any unit tests that would be useful to add for the new residue template logic?

Reply:
Added unit tests to [tests/test_h](https://github.com/proteneer/timemachine/pull/1433/commits/541c80a9b517fe78b56427e406e3b2c21f775cb6#diff-f85453bc1091fce2a19b7e82893ba2f6ae091da78f54049d2e1d7e2ad9900690)